administrative-sdk/speechRecording.js

```diff
@@ -123,197 +102,155 @@ class SpeechRecording {
-      // RPC success callback
-      recordingCb,
-      // RPC error callback
-      function(res) {
-        Connection.logRPCError(res);
-        _ecb(res);
+      function startRecording(recordingId) {
+        connection._recordingId = recordingId;
+        console.log('Got recordingId after initialisation: ' + connection._recordingId);
       }
-    );
-
-    // Stop listening when the audio recorder stopped.
-    var recordedCb = function(activeRecordingId, audioBlob, forcedStop) {
-      connection._session.call('nl.itslanguage.recording.close',
-        [connection._recordingId]).then(
-        // RPC success callback
-        function(res) {
-          console.log(res);
-          // Pass along details to the success callback
-          _cb(res, forcedStop);
-        },
-        // RPC error callback
-        function(res) {
-          Connection.logRPCError(res);
-          _ecb(res);
-        }
-      );
 
-      recorder.removeEventListener('recorded', recordedCb);
-      recorder.removeEventListener('dataavailable', dataavailableCb);
-      // This session is over.
-      connection._recordingId = null;
-    };
-    recorder.addEventListener('recorded', recordedCb);
+      recorder.addEventListener('recorded', recordedCb);
+      connection._session.call('nl.itslanguage.recording.init_recording', [])
+          .then(startRecording)
+          .then(() => {
+            self.speechRecordingInitChallenge(connection, challenge)
+                .then(function() {
+                  var p = new Promise(function(resolve) {
+                    if (recorder.hasUserMediaApproval()) {
+                      resolve();
+                    } else {
+                      recorder.addEventListener('ready', resolve);
+                    }
+                  });
+                  p.then(self.speechRecordingInitAudio(connection, recorder, startStreaming));
+                });
+          },
+            function(res) {
+              Connection.logRPCError(res);
+              errorEncountered(res);
+            }
+          );
+    }
+    );
   }
 
   /**
-   * Callback used by getSpeechRecording.
-   *
-   * @callback Sdk~getSpeechRecordingCallback
-   * @param {its.SpeechRecording} recording Retrieved speech recording domain model instance.
-   */
-  getSpeechRecordingCallback(recording) {}
-
-  /**
-   * Error callback used by getSpeechRecording.
-   *
-   * @callback Sdk~getSpeechRecordingErrorCallback
-   * @param {object[]} errors Array of errors.
-   */
-  getSpeechRecordingErrorCallback(errors) {}
-
-  /**
    * Get a speech recording in a speech challenge.
    *
    * @param {SpeechChallenge} challenge Specify a speech challenge.
    * @param {string} recordingId Specify a speech recording identifier.
    * @param {Sdk~getSpeechRecordingCallback} [cb] The callback that handles the response.
    * @param {Sdk~getSpeechRecordingErrorCallback} [ecb] The callback that handles the error response.
```

Comment:
Missing documentation update. Callbacks are no longer taken and a promise is returned.